src/Bot.hs

```diff
@@ -203,12 +205,21 @@ builtinCommands =
     , ( "wiggle"
       , ( "Wiggle the tenticle (integration with https://github.com/tsoding/wiggle)"
         , transR (Identity . messageSender) $
-          cmapR (URI.encode . T.unpack . senderName) $
-          Reaction $ \(Identity name) -> do
-            request <- parseRequest [qms|http://localhost:8081/wiggle/{name}|]
-            void $ httpRequest request))
+          cmapR (URI.encode . T.unpack . senderName) wiggle))
     ]
 
+wiggle :: Comonad w => Reaction w String
+wiggle = Reaction $ \wa -> wiggle' (extract wa) True
+  where
+    defaultTimeoutMillis = 10000
+    wiggle' name isRetry = do
+      request <- parseRequest [qms|http://localhost:8081/wiggle/{name}|]
+      response <- httpRequest request
+      let status = getResponseStatus response
+      -- TODO: handle error on request when free monad will provide such ability
```

Comment:
But when such ability is going to be implemented? Do we have a separate issue for that as well?

I'd suggest to rephrase that TODO as `TODO: wiggle retry doesn't handle the server being down`. The scope of such TODO includes refactoring both `httpRequest` and `wiggle` functions.